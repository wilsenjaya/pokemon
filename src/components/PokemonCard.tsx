/* eslint-disable react/require-default-props */
import { useRouter } from 'next/router';
import Image from 'next/image';
import {
  Heading,
  Box,
  Text,
  Stack,
  Button,
  Badge,
} from '@chakra-ui/react';

type Props = {
  id: string;
  name: string;
  image: string;
  nickName?: string;
  totalOwned?: number;
  onRemove?: () => void;
};

const PokemonCard = ({
  id,
  name,
  image,
  nickName,
  totalOwned,
  onRemove,
}: Props) => {
  const router = useRouter();

  const onClickDetail = () => router.push(`/pokemon/${id}`);

  return (
    <Box
      p={4}
      boxShadow="xl"
      textAlign="center"
      borderWidth="1px"
      borderRadius="2xl"
      minHeight={300}
    >
      <Image src={image} alt={`${name}_image`} width={150} height={150} />
      <Heading fontSize="2xl" fontFamily="body">
        {name}
      </Heading>
      <Text fontWeight={600} color="gray.500" mb={4}>
        {`# ${id}`}
      </Text>

      {!!nickName && (
        <Heading fontSize="2xl" fontFamily="body">
          {nickName}
        </Heading>
      )}

      <Stack align="center" justify="center" direction="row" mt={4} spacing={3}>
        <Button fontSize="sm" rounded="full" onClick={onClickDetail}>
          Check Detail
        </Button>

        {onRemove && (
          <Button
            fontSize="sm"
            rounded="full"
            colorScheme="red"
            onClick={onRemove}
          >
            Release
          </Button>
        )}
      </Stack>

      {!!totalOwned && (
        <Badge
          mt={4}
          px={2}
          py={1}
          variant="outline"
          fontWeight="400"
          colorScheme={totalOwned > 0 ? 'green' : 'gray'}
        >
          {`Total Owned: ${totalOwned}`}
        </Badge>
      )}
    </Box>
  );
};

export default PokemonCard;
