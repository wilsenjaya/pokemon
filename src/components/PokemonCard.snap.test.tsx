/* eslint-disable react/jsx-props-no-spreading */
import renderer from 'react-test-renderer';

import PokemonCard from './PokemonCard';

describe('#PokemonCard Component', () => {
  const props = {
    id: '1',
    name: 'bulbasaur',
    image: 'https://picsum.photos/200/',
  };

  it('should renders correctly with default props', () => {
    const tree = renderer.create(<PokemonCard {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should renders correctly with optional props', () => {
    const mockProps = {
      ...props,
      nickName: 'bulba',
      totalOwned: 1,
      onRemove: jest.fn(),
    };

    const tree = renderer.create(<PokemonCard {...mockProps} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
