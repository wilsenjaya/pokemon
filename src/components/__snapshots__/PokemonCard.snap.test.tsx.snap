// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#PokemonCard Component should renders correctly with default props 1`] = `
<div
  className="css-utde3r"
>
  <span
    style={
      Object {
        "background": "none",
        "border": 0,
        "boxSizing": "border-box",
        "display": "inline-block",
        "height": "initial",
        "margin": 0,
        "maxWidth": "100%",
        "opacity": 1,
        "overflow": "hidden",
        "padding": 0,
        "position": "relative",
        "width": "initial",
      }
    }
  >
    <span
      style={
        Object {
          "background": "none",
          "border": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": "initial",
          "margin": 0,
          "maxWidth": "100%",
          "opacity": 1,
          "padding": 0,
          "width": "initial",
        }
      }
    >
      <img
        alt=""
        aria-hidden={true}
        src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27150%27%20height=%27150%27/%3e"
        style={
          Object {
            "background": "none",
            "border": 0,
            "display": "block",
            "height": "initial",
            "margin": 0,
            "maxWidth": "100%",
            "opacity": 1,
            "padding": 0,
            "width": "initial",
          }
        }
      />
    </span>
    <img
      alt="bulbasaur_image"
      data-nimg="intrinsic"
      decoding="async"
      onError={[Function]}
      onLoad={[Function]}
      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
      style={
        Object {
          "border": "none",
          "bottom": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": 0,
          "left": 0,
          "margin": "auto",
          "maxHeight": "100%",
          "maxWidth": "100%",
          "minHeight": "100%",
          "minWidth": "100%",
          "objectFit": undefined,
          "objectPosition": undefined,
          "padding": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 0,
        }
      }
    />
    <noscript>
      <img
        alt="bulbasaur_image"
        data-nimg="intrinsic"
        decoding="async"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Fpicsum.photos%2F200%2F&w=384&q=75"
        srcSet="/_next/image?url=https%3A%2F%2Fpicsum.photos%2F200%2F&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fpicsum.photos%2F200%2F&w=384&q=75 2x"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
    </noscript>
  </span>
  <h2
    className="chakra-heading css-178fexg"
  >
    bulbasaur
  </h2>
  <p
    className="chakra-text css-qi3vhq"
  >
    # 1
  </p>
  <div
    className="chakra-stack css-gfbv0r"
  >
    <button
      className="chakra-button css-1ystkvu"
      onClick={[Function]}
      type="button"
    >
      Check Detail
    </button>
  </div>
</div>
`;

exports[`#PokemonCard Component should renders correctly with optional props 1`] = `
<div
  className="css-utde3r"
>
  <span
    style={
      Object {
        "background": "none",
        "border": 0,
        "boxSizing": "border-box",
        "display": "inline-block",
        "height": "initial",
        "margin": 0,
        "maxWidth": "100%",
        "opacity": 1,
        "overflow": "hidden",
        "padding": 0,
        "position": "relative",
        "width": "initial",
      }
    }
  >
    <span
      style={
        Object {
          "background": "none",
          "border": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": "initial",
          "margin": 0,
          "maxWidth": "100%",
          "opacity": 1,
          "padding": 0,
          "width": "initial",
        }
      }
    >
      <img
        alt=""
        aria-hidden={true}
        src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27150%27%20height=%27150%27/%3e"
        style={
          Object {
            "background": "none",
            "border": 0,
            "display": "block",
            "height": "initial",
            "margin": 0,
            "maxWidth": "100%",
            "opacity": 1,
            "padding": 0,
            "width": "initial",
          }
        }
      />
    </span>
    <img
      alt="bulbasaur_image"
      data-nimg="intrinsic"
      decoding="async"
      onError={[Function]}
      onLoad={[Function]}
      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
      style={
        Object {
          "border": "none",
          "bottom": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": 0,
          "left": 0,
          "margin": "auto",
          "maxHeight": "100%",
          "maxWidth": "100%",
          "minHeight": "100%",
          "minWidth": "100%",
          "objectFit": undefined,
          "objectPosition": undefined,
          "padding": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 0,
        }
      }
    />
    <noscript>
      <img
        alt="bulbasaur_image"
        data-nimg="intrinsic"
        decoding="async"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Fpicsum.photos%2F200%2F&w=384&q=75"
        srcSet="/_next/image?url=https%3A%2F%2Fpicsum.photos%2F200%2F&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fpicsum.photos%2F200%2F&w=384&q=75 2x"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
    </noscript>
  </span>
  <h2
    className="chakra-heading css-178fexg"
  >
    bulbasaur
  </h2>
  <p
    className="chakra-text css-qi3vhq"
  >
    # 1
  </p>
  <h2
    className="chakra-heading css-178fexg"
  >
    bulba
  </h2>
  <div
    className="chakra-stack css-gfbv0r"
  >
    <button
      className="chakra-button css-1ystkvu"
      onClick={[Function]}
      type="button"
    >
      Check Detail
    </button>
    <button
      className="chakra-button css-1ystkvu"
      onClick={[MockFunction]}
      type="button"
    >
      Release
    </button>
  </div>
  <span
    className="chakra-badge css-wo20ao"
  >
    Total Owned: 1
  </span>
</div>
`;
